src/clientSideScene/segments.ts

```diff
@@ -925,6 +952,584 @@ class CircleThreePointSegment implements SegmentUtils {
 }
 
+class ArcSegment implements SegmentUtils {
+  init: SegmentUtils['init'] = ({
+    prevSegment,
+    input,
+    id,
+    pathToNode,
+    isDraftSegment,
+    scale = 1,
+    theme,
+    isSelected,
+    sceneInfra,
+  }) => {
+    if (input.type !== 'arc-segment') {
+      return new Error('Invalid segment type')
+    }
+    const { from, to, center, radius, ccw } = input
+    const baseColor = getThemeColorForThreeJs(theme)
+    const color = isSelected ? 0x0000ff : baseColor
+
+    // Calculate start and end angles
+    const startAngle = Math.atan2(from[1] - center[1], from[0] - center[0])
+    const endAngle = Math.atan2(to[1] - center[1], to[0] - center[0])
+
+    const group = new Group()
+    const geometry = createArcGeometry({
+      center,
+      radius,
+      startAngle,
+      endAngle,
+      ccw,
+      isDashed: isDraftSegment,
+      scale,
+    })
+    const mat = new MeshBasicMaterial({ color })
+    const arcMesh = new Mesh(geometry, mat)
+    const meshType = isDraftSegment ? ARC_SEGMENT_DASH : ARC_SEGMENT_BODY
+
+    // Create handles for the arc
+
+    const endAngleHandle = createArrowhead(scale, theme, color)
+    endAngleHandle.name = ARC_ANGLE_END
+    endAngleHandle.userData.type = ARC_ANGLE_END
+
+    const circleCenterGroup = createCircleCenterHandle(scale, theme, color)
+
+    // A radius indicator that appears from the center to the perimeter
+    const radiusIndicatorGroup = createLengthIndicator({
+      from: center,
+      to: from,
+      scale,
+    })
+
+    // Create a line from the center to the 'to' point
+    const centerToFromLine = createLine({
+      from: center,
+      to: from,
+      scale,
+      color: 0xaaaaaa, // Light gray color for the line
+    })
+    centerToFromLine.name = ARC_CENTER_TO_FROM
+    const centerToToLine = createLine({
+      from: center,
+      to,
+      scale,
+      color: 0xaaaaaa, // Light gray color for the line
+    })
+    centerToToLine.name = ARC_CENTER_TO_TO
+    const angleReferenceLine = createLine({
+      // from: center,
+      from: [center[0] + 28 * scale, center[1]],
+      to: [center[0] + 32 * scale, center[1]],
+      scale,
+      color: 0xaaaaaa, // Light gray color for the line
+    })
+    angleReferenceLine.name = ARC_ANGLE_REFERENCE_LINE
+
+    // Create a curved line with an arrow to indicate the angle
+    const angleIndicator = createAngleIndicator({
+      center,
+      radius: radius / 2, // Half the radius for the indicator
+      startAngle: 0,
+      endAngle,
+      scale,
+      color: 0xaaaaaa, // Red color for the angle indicator
+    }) as Line
+    angleIndicator.name = 'angleIndicator'
+
+    // Create a new angle indicator for the end angle
+    const endAngleIndicator = createAngleIndicator({
+      center,
+      radius: radius / 2, // Half the radius for the indicator
+      startAngle: 0,
+      endAngle: (endAngle * Math.PI) / 180,
+      scale,
+      color: 0xaaaaaa, // Green color for the end angle indicator
```

Comment:
How is this green?

Reply:
Sorry stale comment 3791d92defc11944f283c9cacf5292b2609c6675